File: JUNE-2025/30-06-2025/p594.swift
class Solution {
    func findLHS(_ nums: [Int]) -> Int {
        let freq = [Int:Int](nums.map{num in (num,1)}, uniquingKeysWith:+)
        var lhsCount = 0
        for (key,value) in freq where freq[key+1] != nil {
            let count = value+freq[key+1]!
            lhsCount = max(lhsCount, count)
        }
        return lhsCount
    }
}

File: JUNE-2025/30-06-2025/p594.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
function findLHS($nums) {
    $freq = array_count_values($nums);
    

    $maxLen = 0;
    foreach ($freq as $num => $count) {
        if (isset($freq[$num + 1])) {
            $maxLen = max($maxLen, $count + $freq[$num + 1]);
        }
    }

    return $maxLen;
}

}